Simplify variant method to use fallback with error handling

hahwul opened the pull request:
This PR addresses #12 by adding error handling for cases where the variant setting is missing in the Jekyll config.

🔍 Issue
When variant is not specified in _config.yml, the Jekyll build process fails due to an undefined method error.

✅ Changes
* Added a fallback mechanism to handle missing variant values.

📌 Notes
* This change ensures that Jekyll can still build successfully even if variant is not explicitly defined.
* The default behavior follows the existing logic while preventing unexpected crashes.

Let me know if any modifications are needed! 🚀

hahwul committed:
Simplify variant method to use fallback with error handling

Signed-off-by: HAHWUL <hahwul@gmail.com>

skatkov commented on lib/jekyll-heroicons.rb:
```diff
@@ -63,13 +63,9 @@ def prepare(markup)
-      else
-        DEFAULT_VARIANT
-      end
+      @options[:variant] || config["variant"] || DEFAULT_VARIANT
```
**suggestion:** Nice improvement! I love it that code looks simpler this way.


But we're swallowing error here - it doesn't seem nice ;-)

How about an alternative approach, e.g.:
```
@options&.dig(:variant) || config&.dig("variant") || DEFAULT_VARIANT
```

so we can remove `rescue`, the error swallowing part?

hahwul replied:
I agree! That approach looks cleaner and avoids swallowing the error. I've already updated it. However, removing rescue causes the same issue in my environment. It seems like an environment-specific problem, so I'll investigate further!

skatkov replied:
@hahwul 
this would resolve it, notice a `&.` it will actually not raise an error if config is nil.

`@options&.dig(:variant) || config&.dig("variant") || DEFAULT_VARIANT`

hahwul replied:
Oh, I apologize for that. It works well with the code you modified. I made a mistake because I was only thinking about simply deleting the rescue.

skatkov commented:
Can you also add a line into a CHANGELOG.md file about this change? It would be greatly appreciated.

hahwul committed:
Remove error handling from variant method and simplify fallback logic

Signed-off-by: HAHWUL <hahwul@gmail.com>

hahwul commented:
## Env
- Jekyll 4.4.1
- Ruby 3.4.1
- jekyll-heroicons-0.4.1

## Devices
- [x] MacBook Air
- [x] MacBook Pro

## Ruby version check
- [x] 3.4.1
- [x] 3.3.7
- [x] 3.3.5
- [x] 3.3.0 

## Heroicon gem
- [x] 0.4.1
- [x] 0.4.0 
- [x] 0.3.1 
- [x] 0.3.0
- [x] 0.2.1 


## Test - Check error case
Make Jekyll App

```bash
jekyll new heroicon_test
cd heroicon_test
```

Add gem (`Gemfile`)

```ruby
gem 'jekyll-heroicons'
```

and bundle

Add plugin (`config.yml`)

```ruby
plugins:
  - jekyll-feed
  - jekyll-heroicons
```

Add heroicon (`index.markdown`)

```ruby
---
# Feel free to add content and custom Front Matter to this file.
# To modify the layout, see https://jekyllrb.com/docs/themes/#overriding-theme-defaults

layout: home
---

{% heroicon bell %}
```

Run Jekyll
```bash
bundle exec jekyll serve
```

```
Configuration file: /Users/pluto/test/heroicon_test/_config.yml
  Liquid Exception: undefined method '[]' for nil in index.markdown
                    ------------------------------------------------
      Jekyll 4.4.1   Please append `--trace` to the `serve` command
                     for any additional information or backtrace.
                    ------------------------------------------------
/Users/pluto/.rvm/gems/ruby-3.4.1/gems/jekyll-heroicons-0.4.1/lib/jekyll-heroicons.rb:68:in 'Jekyll::Heroicons#variant': undefined method '[]' for nil (NoMethodError)

      elsif config["variant"]
                  ^^^^^^^^^^^
        from /Users/pluto/.rvm/gems/ruby-3.4.1/gems/jekyll-heroicons-0.4.1/lib/jekyll-heroicons.rb:60:in 'Jekyll::Heroicons#prepare'
        from /Users/pluto/.rvm/gems/ruby-3.4.1/gems/jekyll-heroicons-0.4.1/lib/jekyll-heroicons.rb:36:in 'Jekyll::Heroicons#initialize'
```

hahwul commented:
@skatkov 
I tested multiple versions of the Ruby and Heroicon gems on two different devices, and I encountered the same error in both cases. If you create a new project and omit the necessary config, do you run into the same error as I do? I'm wondering if this issue is specific to me.

Oh, and if the error isn't handled with `rescue`, it might be helpful to include a note about this case in the README.

skatkov commented on lib/jekyll-heroicons.rb:
```diff
@@ -63,13 +63,7 @@ def prepare(markup)
-      else
-        DEFAULT_VARIANT
-      end
+      @options[:variant] || config["variant"] || DEFAULT_VARIANT
```
Can you try this code?

```suggestion
      @options&.dig(:variant) || config&.dig("variant") || DEFAULT_VARIANT
```

skatkov committed:
Update lib/jekyll-heroicons.rb

skatkov commented:
Committed a small change to your PR. Can you pull changes and try again?

hahwul commented:
@skatkov 
I tried again with the changes you made, and it works well. Thank you so much!

hahwul committed:
Update CHANGELOG.md

hahwul commented:
@skatkov 
I've added the content to the CHANGELOG.md file. I was going to write it for version 0.4.2, but since it's not released yet, I added it under the 'Unreleased' section. Please let me know if there's anything else I should do!

skatkov commented:
Nice!

Thanks for reporting this issue and a Pull Request.
